### PR TITLE
Json validation and fix

### PR DESCRIPTION
**Pull request by s106916**
## Description
1.  This PR consolidates json string validation and loading from index, report and llm three parts into one piece.
2.  including trim extract description from llm result, replace or convert unfit chars, using json_repair for the finally repair before json loading. wrapped all functions into try_parse_json_object() as unique one function to use.     
3. There are two duplicated json_clean_up functions in both index and llm section, consolidate into one under llm model.
4. Furthermore, wrap json_clean_up into try_parse_json_object() function without Exception for friendly retry 

## Proposed Changes
add json-repair

## Checklist

- [x] I have tested these changes locally.
- [x] I have reviewed the code changes.
- [x] I have updated the documentation (if necessary).
- [ ] I have added appropriate unit tests (if applicable).

## Additional Notes

[Add any additional notes or context that may be helpful for the reviewer(s).]

### CODE REVIEW

**Comment by s106916** [comment]
this is fruther consolidate version for 
JSON parsing: always fix all incoming json when operating _manual_json mode #551
https://github.com/microsoft/graphrag/pull/551

**Comment by s106916** [comment]
Hi @AlonsoGuevara, Could you please review it? Happy to work on it and taken any suggestion.

**Comment by s106916** [comment]
rebased from main as requested

File: graphrag/llm/openai/utils.py [hunk 1]
```diff
@@ -87,17 +88,46 @@ def get_completion_llm_args(
...
+    )
+
+    # Remove JSON Markdown Frame
+    if input.startswith("```json"):
+        input = input[len("```json") :]
+    if input.endswith("```"):
+        input = input[: len(input) - len("```")]
+
+    """Fixup potentially malformed json string using json_repair."""
+    input = str(repair_json(json_str=input, return_objects=False))
```
Comment (jgbradley1) [inline graphrag/llm/openai/utils.py:118]: Before calling out to another library via `repair_json` to fix what could be a valid json string at this point, can you add in an attempt to call `json.loads(json_str=input)` before this line...and only call `repair_json()` if the string is still invalid? String operations can be slow at large scale so we want to minimize them where possible.

Comment (s106916) [reply graphrag/llm/openai/utils.py]: add try json.loads() block before repair_json
Many thanks @jgbradley1

**Comment by s106916** [comment]
merge as #801

**Pull request closed without merging**